Mathlib/SetTheory/Cardinal/SchroederBernstein.lean

```diff
@@ -86,41 +86,37 @@ variable {ι : Type u} (β : ι → Type v)
 
 /-- `sets β` -/
 private abbrev sets :=
-  { s : Set (∀ i, β i) | ∀ x ∈ s, ∀ y ∈ s, ∀ (i), (x : ∀ i, β i) i = y i → x = y }
+  { s : Set (∀ i, β i) | ∀ i : ι, s.InjOn fun x => x i }
 
 /-- The cardinals are well-ordered. We express it here by the fact that in any set of cardinals
 there is an element that injects into the others.
 See `Cardinal.conditionallyCompleteLinearOrderBot` for (one of) the lattice instances. -/
 theorem min_injective [I : Nonempty ι] : ∃ i, Nonempty (∀ j, β i ↪ β j) :=
   let ⟨s, hs⟩ := show ∃ s, Maximal (· ∈ sets β) s by
     refine zorn_subset _ fun c hc hcc ↦
-      ⟨⋃₀ c, fun x ⟨p, hpc, hxp⟩ y ⟨q, hqc, hyq⟩ i hi ↦ ?_, fun _ ↦ subset_sUnion_of_mem⟩
-    exact (hcc.total hpc hqc).elim (fun h ↦ hc hqc x (h hxp) y hyq i hi)
-      fun h ↦ hc hpc x hxp y (h hyq) i hi
+      ⟨⋃₀ c, fun i x ⟨p, hpc, hxp⟩ y ⟨q, hqc, hyq⟩ hi ↦ ?_, fun _ ↦ subset_sUnion_of_mem⟩
+    exact (hcc.total hpc hqc).elim (fun h ↦ hc hqc i (h hxp) hyq hi)
+      fun h ↦ hc hpc i hxp (h hyq) hi
   let ⟨i, e⟩ :=
-    show ∃ i, ∀ y, ∃ x ∈ s, (x : ∀ i, β i) i = y from
+    show ∃ i, Surjective fun x : s => x.val i from
       Classical.by_contradiction fun h =>
         have h : ∀ i, ∃ y, ∀ x ∈ s, (x : ∀ i, β i) i ≠ y := by
-          simpa only [ne_eq, not_exists, not_forall, not_and] using h
+          simpa only [ne_eq, Surjective, Subtype.exists, exists_prop, not_exists, not_forall,
```

Comment:
Can this be replaced by `simpa using h`?

Reply:
It can be replaced with `simpa [Surjective] using h`. I'm not sure which is the best style though.

Reply:
To my understanding `simpa only` was used more in Mathlib 3 where performance was a greater bottleneck, but is now only really used for non-terminal simps. Someone else can perhaps corroborate on that.

Reply:
Okay. It's changed